Heat_method_3/doc/Heat_method_3/Heat_method_3.txt

```diff
@@ -0,0 +1,249 @@
+`boost::make_assoc_property_map()`, which generates a vertex distance property map.
+
+\cgalExample{Heat_method_3/heat_method_polyhedron.cpp}
+
+
+\subsection HM_example_Class Using the Heat Method Class
+
+The following example shows the heat method class. It can be used
+when one adds and removes source vertices. It precomputes matrix
+factorization, which depend only on the input mesh and not the particular
+set of source vertices.  In the example we compute the distances to one
+source, add the farthest vertex as a second source vertex, and then compute
+the distances with respect to these two sources.
+
+\cgalExample{Heat_method_3/heat_method_surface_mesh.cpp}
+
+
+\subsection HM_example_Intrinsic Switching off the Intrinsic Delaunay Triangulation
+
+The following example shows the heat method on a triangle mesh without using the
+intrinsic Delaunay triangulation (iDT) algorithm, for example because by constrution
+your meshes have a good quality (Poor quality in this case means that the input
+is far from Delaunay, though even in this case one may still get good results without iDT,
+depending on the specific geometry of the surface.)
+
+\cgalExample{Heat_method_3/heat_method_surface_mesh_direct.cpp}
+
+
+
+\section sec_HM_definitions Theoretical Background
+
+Section \ref Subsection_HM_Definitions_Intro gives an overview of the theory needed by the Heat method.
+Section \ref Subsection_HM_IDT_Definitions gives the background needed for the Intrinsic Delaunay triangulation.
+
+\subsection Subsection_HM_Definitions_Intro The Heat Method Algorithm
+
+For a detailed overview of the heat method, the reader may consult
+\cgalCite{cgal:cww-ghnac-13} to read the original article. In the
+sequel, we introduce the basic notions so as to explain our
+algorithms. In general, the heat method is applicable to any setting
+if there exists a gradient operator \f$ \nabla\f$, a divergence
+operator \f$\nabla\f$ and a Laplace operator \f$\Delta = \nabla \cdot
+\nabla\f$, which are standard derivatives from vector calculus.
+
+The Heat Method consists of three main steps:
+           -# Integrate the heat flow \f$ \dot u = \Delta u\f$ for some fixed time \f$t\f$.
+           -# Evaluate the vector field   \f$ X = -\nabla u_t / |\nabla u_t| \f$.
+           -# Solve the Poisson Equation \f$ \Delta \phi = \nabla \cdot X \f$.
+
+
+The function \f$ \phi \f$ is an approximation of the distance to the given source set and approaches the true distance as t goes to zero.
+The algorithm must then be translated in to a discrete algorithm by replacing the derivatives in space and time with approximations.
+
+The heat equation can be discretized in time using a single backward Euler step. This means the following equation must be solved:
+
+\f$(id-t\Delta)u_t = \delta(x) \f$ where \f$\delta(x)\f$ is a Dirac delta encoding an 'infinite' spike of heat (1 if x is in the source set, 0 otherwise)
```

Comment:
single quotes?

Reply:
Single quotes seem fine. Honestly we start cutting hairs here.

Reply:
I changed it to "

Reply:
You don't need to say "I did this" because I'll be able to look at the diff + nothing has been changed anyway.

Reply:
I pushed it 21h ago.